server/controllers/feeds.ts

```diff
@@ -211,126 +260,357 @@ function initFeed (parameters: {
+        })
       })
-    }
 
-    const localLink = WEBSERVER.URL + video.getWatchStaticPath()
+      const sortedVideos = orderBy(preferredVideos, [ 'bitrate' ], [ 'desc' ])
 
-    feed.addItem({
-      title: video.name,
-      id: localLink,
-      link: localLink,
-      description: mdToOneLinePlainText(video.getTruncatedDescription()),
-      content: toSafeHtml(video.description),
-      author: [
-        {
-          name: video.VideoChannel.getDisplayName(),
-          link: video.VideoChannel.Actor.url
-        }
-      ],
-      date: video.publishedAt,
-      nsfw: video.nsfw,
-      torrents,
-
-      // Enclosure
-      video: videoFiles.length !== 0
-        ? {
-          url: videoFiles[0].url,
-          length: videoFiles[0].fileSize,
-          type: videoFiles[0].type
-        }
-        : undefined,
-
-      // Media RSS
-      videos: videoFiles,
-
-      embed: {
-        url: WEBSERVER.URL + video.getEmbedStaticPath(),
-        allowFullscreen: true
-      },
-      player: {
-        url: WEBSERVER.URL + video.getWatchStaticPath()
-      },
-      categories,
-      community: {
-        statistics: {
-          views: video.views
+      const streamingPlaylists = video.VideoStreamingPlaylists
+        .filter((streamingPlaylist) => streamingPlaylist.type === VideoStreamingPlaylistType.HLS)
+        .map(streamingPlaylist => {
+          const result = {
+            type: 'application/x-mpegURL',
+            title: 'HLS',
+            sources: [
+              { uri: streamingPlaylist.getMasterPlaylistUrl(video) }
+            ]
+          }
+
+          if (video.language) Object.assign(result, { language: video.language })
+
+          return result
+        })
+
+      const media = [ ...sortedVideos, ...streamingPlaylists ].filter(m => m)
+
+      let category: string
+      if (video.category) {
+        category = getCategoryLabel(video.category)
+      }
+
+      const videoCaptions = await VideoCaptionModel.listVideoCaptions(video.id)
+
+      const captions = videoCaptions?.map(caption => {
+        const type = MIMETYPES.VIDEO_CAPTIONS.EXT_MIMETYPE[extname(caption.filename)]
+        if (!type) return null
+        return {
+          url: caption.getFileUrl(video),
+          language: caption.language,
+          type,
+          rel: 'captions'
         }
-      },
-      thumbnails: [
-        {
-          url: WEBSERVER.URL + video.getPreviewStaticPath(),
-          height: PREVIEWS_SIZE.height,
-          width: PREVIEWS_SIZE.width
+      }).filter(c => c)
+
+      const customTags: CustomTag[] = await Hooks.wrapObject(
+        [],
+        'filter:api.feed.podcast.item.custom-tags.result',
+        { video }
+      )
+
+      const author = {
+        name: video.VideoChannel.Account.getDisplayName(),
+        href: video.VideoChannel.Account.Actor.url
+      }
+      const item = {
```

Comment:
Please refactor this big object creation inside a function and reuse it to generate live JSON too